woocommerce/woocommerce
# Update cache action and fix hashFiles bug

### Submission Review Guidelines:

### Changes proposed in this Pull Request:

This adds 2 simple changes to fix https://github.com/woocommerce/woocommerce/issues/37745

1. This updates the cache action used in the monorepo setup GH action to use the latest released version. There have been many bug fixes and updates since this action was last updated. This is not needed to fix the problem, but I wanted to update the action while I was working in this area.

2. Fixes a bug where the combination of `path` and matcher passed to `hashFiles` resulted in the hash being empty and there was no cache restoration happening. `hashFiles` will return an empty string whenever the matcher matches no files.

To confirm the fix, I pushed a change that should not bust the cache after the first commit.

Now you can see the output being correct and that the cache is being restored from a result of hashing the turbo meta.json files.

e.g. I saw this output for 2 runs of the CI:

First run of CI that generated the now correct cache key:

`key: Linux-build-output-07f6725759c35c31ec851eeede02b20ee638aa198b7e1716bbd2b3d2d74265c9`

Second run, restored the cache due to their being no changes:

`Cache restored from key: Linux-build-output-07f6725759c35c31ec851eeede02b20ee638aa198b7e1716bbd2b3d2d74265c9`



### How to test the changes in this Pull Request:

1. It is sufficient to see that all the Github CI checks pass and to see that the "WooCommerce MonoRepo Setup" steps used in many of the checks is successfully restoring the cache with output like: `Cache restored from key: Linux-build-output-07f6725759c35c31ec851eeede02b20ee638aa198b7e1716bbd2b3d2d74265c9`
